test/kubernetes_unittest.cc

```diff
@@ -68,4 +80,67 @@ TEST_F(KubernetesTest, GetNodeMetadata) {
   });
   EXPECT_EQ(big->ToString(), m.metadata.metadata->ToString());
 }
+
+TEST_F(KubernetesTest, ComputePodAssociations) {
+  const std::string api_version = "1.2.3";
+  const std::string kind = "TestKind";
+  const std::string uid1 = "TestUID1";
+  Configuration config(std::stringstream(
+    "KubernetesClusterName: TestClusterName\n"
+    "KubernetesClusterLocation: TestClusterLocation\n"
+    "MetadataIngestionRawContentVersion: TestVersion\n"
+    "InstanceZone: TestZone\n"
+    "InstanceId: TestID\n"
+  ));
+  Environment environment(config);
+  KubernetesReader reader(config, nullptr);  // Don't need HealthChecker.
+  const std::string encoded_ref = boost::algorithm::join(
+      std::vector<std::string>{api_version, kind, uid1}, "/");
```

Comment:
Let's avoid this kind of logic in tests. Anything more complicated than "the same string goes in and out" is probably not worth the readability hit. Just use literal strings.